middle_end/flambda2/simplify/env/data_flow.ml

```diff
@@ -544,7 +544,11 @@ module Dependency_graph = struct
       match (used_closure_vars : _ Or_unknown.t) with
       | Unknown -> fun _ -> true
       | Known used_closure_vars ->
-        Name_occurrences.mem_closure_var used_closure_vars
+        fun closure_var ->
+          (not
+             (Compilation_unit.is_current
+                (Var_within_closure.get_compilation_unit closure_var)))
+          || Name_occurrences.mem_closure_var used_closure_vars closure_var
```

Comment:
This chunk of code is repeated in a few places (e.g. the `either not in the current compilation unit, or in the given set of used closure vars`), maybe it could be worth it to factor out that code in a function called in all those places. Maybe it would also help future code to also integrate this check ?

Reply:
I could move the check inside `Name_occurrences.mem_closure_var`, but that feels weird.
The most coherent thing would be to add a function `is_closure_var_used` to the relevant environments, but that still leaves a lot of duplication (at least `Downwards_acc`, `Upwards_acc` and `To_cmm_env` would need this code).
Maybe a new `closure_var_is_used_or_imported` function to `Name_occurrences` would be best ?